ethereum/statetransition/src/test/java/tech/pegasys/teku/statetransition/forkchoice/MergeTransitionConfigCheckTest.java

```diff
@@ -0,0 +1,161 @@
+import tech.pegasys.teku.spec.Spec;
+import tech.pegasys.teku.spec.TestSpecFactory;
+import tech.pegasys.teku.spec.config.SpecConfigBuilder.BellatrixBuilder;
+import tech.pegasys.teku.spec.config.SpecConfigLoader;
+import tech.pegasys.teku.spec.executionengine.ExecutionEngineChannel;
+import tech.pegasys.teku.spec.executionengine.TransitionConfiguration;
+import tech.pegasys.teku.spec.logic.common.block.AbstractBlockProcessor;
+import tech.pegasys.teku.spec.util.DataStructureUtil;
+
+class MergeTransitionConfigCheckTest {
+
+  private static final UInt64 BELLATRIX_FORK_EPOCH = UInt64.ONE;
+  private static final Bytes32 TERMINAL_BLOCK_HASH = Bytes32.random();
+  private static final UInt64 TERMINAL_BLOCK_EPOCH = UInt64.valueOf(2);
+
+  private final ExecutionEngineChannel executionEngine = mock(ExecutionEngineChannel.class);
+  private final StubTimeProvider timeProvider = StubTimeProvider.withTimeInSeconds(10_000);
+  private final StubAsyncRunner asyncRunner = new StubAsyncRunner(timeProvider);
+  private final EventLogger eventLogger = Mockito.mock(EventLogger.class);
+
+  private DataStructureUtil dataStructureUtil;
+  private MergeTransitionConfigCheck mergeTransitionConfigCheck;
+  private TransitionConfiguration localTransitionConfiguration;
+
+  @BeforeAll
+  public static void initSession() {
+    AbstractBlockProcessor.blsVerifyDeposit = false;
+  }
+
+  @AfterAll
+  public static void resetSession() {
+    AbstractBlockProcessor.blsVerifyDeposit = true;
+  }
+
+  @Test
+  void shouldDetectTerminalConfigProblems() {
+    setUpTerminalBlockHashConfig();
+    final UInt256 wrongRemoteTTD = dataStructureUtil.randomUInt256();
+    final Bytes32 wrongRemoteTBH = dataStructureUtil.randomBytes32();
+
+    TransitionConfiguration wrongRemoteConfig;
+
+    assertThat(mergeTransitionConfigCheck.start()).isCompleted();
+
+    // wrong TTD
+    wrongRemoteConfig =
+        new TransitionConfiguration(
+            wrongRemoteTTD, localTransitionConfiguration.getTerminalBlockHash(), UInt64.ZERO);
+    when(executionEngine.exchangeTransitionConfiguration(localTransitionConfiguration))
+        .thenReturn(SafeFuture.completedFuture(wrongRemoteConfig));
+
+    asyncRunner.executeQueuedActions();
+
+    verify(eventLogger)
+        .transitionConfigurationTtdTbhMismatch(
+            localTransitionConfiguration.toString(), wrongRemoteConfig.toString());
+
+    // wrong TBH
+    wrongRemoteConfig =
+        new TransitionConfiguration(
+            localTransitionConfiguration.getTerminalTotalDifficulty(), wrongRemoteTBH, UInt64.ZERO);
+    when(executionEngine.exchangeTransitionConfiguration(localTransitionConfiguration))
+        .thenReturn(SafeFuture.completedFuture(wrongRemoteConfig));
+
+    asyncRunner.executeQueuedActions();
+
+    verify(eventLogger)
+        .transitionConfigurationTtdTbhMismatch(
+            localTransitionConfiguration.toString(), wrongRemoteConfig.toString());
+
+    // remote TBH TBN inconsistency
```

Comment:
these do sound like we're lumping a stack of tests into one... should they be separate?